pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/AuxClassPathController.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+
+package net.sourceforge.pmd.util.fxdesigner.util;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+import net.sourceforge.pmd.util.fxdesigner.DesignerRoot;
+import net.sourceforge.pmd.util.fxdesigner.MainDesignerController;
+import net.sourceforge.pmd.util.fxdesigner.util.beans.SettingsOwner;
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.stage.FileChooser;
+import javafx.stage.Stage;
+
+public class AuxClassPathController implements Initializable, SettingsOwner {
+
+    private final DesignerRoot designerRoot;
+
+    @FXML
+    private Button removeFiles;
+
+    @FXML
+    private TableView fileTable;
+
+    @FXML
+    private Button selectFile;
+
+    @FXML
+    private TableColumn fileList;
+
+    @FXML
+    private TableColumn fileAdd;
+
+    @FXML
+    private final MainDesignerController parent;
+
+
+    public AuxClassPathController(DesignerRoot designerRoot, MainDesignerController parent) {
+        this.designerRoot = designerRoot;
+        this.parent = parent;
+    }
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+
+        selectFile.setOnAction(e -> fileSeleced());
+
+    }
+
+
+    private void fileSeleced() {
+        FileChooser chooser = new FileChooser();
```

Comment:
You should add an extension filter to only allow selecting jar files. There's an example [here](https://docs.oracle.com/javase/8/javafx/api/javafx/stage/FileChooser.html).